### PR TITLE
Preselect template for user template provision source

### PR DESCRIPTION
**Pull request by rawagner**

### CODE REVIEW

File: src/components/Wizards/CreateVmWizard/BasicSettingsTab.js [hunk 1]
```diff
@@ -199,6 +199,18 @@ const publish = ({ basicSettings, namespaces, templates, selectedNamespace, onCh
     [target]: value
   };
 
+  if (target === 'imageSourceType' && value.value === PROVISION_SOURCE_TEMPLATE) {
+    const currentUserTemplate = get(newBasicSettings.userTemplate, 'value');
+    if (!currentUserTemplate) {
+      const allTemplates = getTemplate(templates, TEMPLATE_TYPE_VM);
+      if (allTemplates.length > 0) {
+        newBasicSettings.userTemplate = {
```
Comment (suomiy) [inline src/components/Wizards/CreateVmWizard/BasicSettingsTab.js]: you can use asValueObject function here

File: src/components/Wizards/CreateVmWizard/BasicSettingsTab.js [hunk 2]
```diff
@@ -199,6 +199,18 @@ const publish = ({ basicSettings, namespaces, templates, selectedNamespace, onCh
     [target]: value
   };
 
+  if (target === 'imageSourceType' && value.value === PROVISION_SOURCE_TEMPLATE) {
```
Comment (suomiy) [inline src/components/Wizards/CreateVmWizard/BasicSettingsTab.js]: please make a constant out of this

File: src/components/Wizards/CreateVmWizard/tests/BasicSettingsTab.test.js [hunk 3]
```diff
@@ -167,4 +167,24 @@ describe('<BasicSettingsTab />', () => {
...
+
+    expectMockToBeCalledWith(
+      onChange,
+      {
+        ...validBasicSettings,
+        imageSourceType: {
+          value: PROVISION_SOURCE_TEMPLATE
+        },
+        userTemplate: {
+          value: 'linux-template'
```
Comment (suomiy) [inline src/components/Wizards/CreateVmWizard/tests/BasicSettingsTab.test.js]: could be the name of the first user template instead of the string

**Comment by rawagner** [comment]
@suomiy fixed all

**Pull request merged**